tests/test_runs/test_run.py

```diff
@@ -179,3 +183,31 @@ def test_to_from_filesystem_no_model(self):
         # assert default behaviour is throwing an error
         with self.assertRaises(ValueError, msg='Could not find model.pkl'):
             openml.runs.OpenMLRun.from_filesystem(cache_path)
+
+    def test_publish_with_local_loaded_flow(self):
```

Comment:
Do you think it would make sense to add a test which downloads a flow from the internet, caches it, does a run, and uploads the run afterwards? Or is the functionality used more or less the same?

Reply:
It's the same for running the experiment, but it's different at upload time (because the flow already exists). Added [this test](https://github.com/openml/openml-python/blob/fix498/tests/test_runs/test_run_functions.py#L1508).